jupyterhub/templates/hub/deployment.yaml

```diff
@@ -4,6 +4,19 @@ metadata:
   name: hub
 spec:
   replicas: 1
+  strategy:
+    rollingUpdate:
+      {{ if eq .Values.hub.db.type "sqlite-pvc" }}
+      # If we're using a PVC, set maxUnavailable to 1
+      # This is required, since the pod attached to PVC must die
+      # before new pod can run.
+      maxUnavailable: 1
+      {{ else }}
+      # If we aren't using a PVC, it's ok to spawn up a new pod
+      # before killing the currently running one. This also makes the '--wait'
+      # flag of helm work properly
+      maxUnavailable: 0
```

Comment:
This may not be okay for postgres/mysql databases. Starting a new pod while the other is shutting down could potentially result in multiple Hubs talking to the same db and produce inconsistent state. It's likely only safe for the case where db.type is null for in-memory-only.